File: composite/type_fichier/FichierPdf.java
package type_fichier;

import fichier.Fichier;

public class FichierPdf extends Fichier {
    
    private String type;

    public FichierPdf(String nom) {
        super(nom);

        this.type = "pdf";
    }

    @Override
    public String getType(){
        return type;
    }

    @Override
    public void decrire() { 
        System.out.println("Nom: " + super.getNom() + "\nType: " + type);
    }
}

File: 2-pattrons-de-structures/composite/code-typescript/type_fichier/fichierPdf.ts
import { Fichier } from "../fichier/fichier.js";

export class FichierPdf extends Fichier {
  constructor(protected nom: string, private type: string = "pdf") {
    super(nom);
  }

  getType(): string {
    return this.type;
  }

  decrire(): void {
    console.log(`Nom: ${this.getNom()}, Type: ${this.getType()}`);
  }
}
